### ISSUE TITLE
Suggestion: CSS selector assertion

### ISSUE THREAD

**Issue by DannyBen** [Orange-OpenSource/hurl#842]
Is there any chance this lovely tool will have a css selector in addition to the xpath selector?

**Comment by jcamiel** [comment]
Yes! Obviously on our radar, we need to think about the syntax but it could be a nice addition to XPath and JSONPath. On the roadmap!

**Comment by DannyBen** [comment]
Thanks for the reply - closing.

**Comment by jcamiel** [comment]
We can let it opened just to be sure to address it. Thanks for the feedback

**Comment by Lewiscowles1986** [comment]
So thank you for re-opening. I Came to ask about similar, but for capturing data; and I have a slightly more specific request from my existing curl + jq + yq + xq + pug scripts.

What are those 2 letter things above? https://blog.lazy-evaluation.net/posts/linux/jq-xq-yq.html

so...

- jq, lots of folks know, it is a command-line util useful for querying json
- yq less people know, it's the same idea, command-line util for querying yaml
- xq even less people know, it's the same idea, command-line util for querying xml (supporting x-path)

Now pug is that, but with CSS Query selector support.

I currently have scripts using these with curl (and without).

A Specific thing I've used these for recently is solving a request chain, where I needed to extract some HTML, from a JSON field, which I needed to complete the next hop in a request chain.

I agree it's an out-there use-case. Maybe it's separate to assertions, but I think the ability to combine these features, speaks to really powerful use-cases, especially where you don't control the vendor producing data, or you've been given an unchangeable spec by a stakeholder that really wants to inappropriately combine technologies.

I'd be happy to test any work on this and can provide dummy API's from real-world where several of these tools can be used together.

There will always be regex; but XML and HTML are not always well-composed as regex, and I'd sooner not add a denial of service attack, many report difficulty understanding, into scripts.

**Comment by esamattis** [comment]
For kinda prior art checkout `htmlq`. I've used it with plain curl to achieve something like hurl but it would be really nice to see something like it build into hurl. 

https://github.com/mgdm/htmlq
